crates/gitbutler-branch-actions/src/status.rs

```diff
@@ -245,6 +259,83 @@ pub fn get_applied_status_cached(
 }
 
 fn compute_locks(
+    ctx: &CommandContext,
+    workspace_head: &git2::Oid,
+    target_sha: &git2::Oid,
+    base_diffs: &BranchStatus,
+    stacks: &Vec<Stack>,
+) -> Result<HashMap<HunkHash, Vec<HunkLock>>> {
+    let repo = ctx.repository();
+    let base_commit = repo.find_commit(*target_sha)?;
+    let workspace_commit = repo.find_commit(*workspace_head)?;
+
+    let diff = &ctx.repository().diff_tree_to_tree(
+        Some(&base_commit.tree()?),
+        Some(&workspace_commit.tree()?),
+        None,
+    )?;
+
+    let files_touched_by_commits = diff
+        .deltas()
+        .filter_map(|d| d.new_file().path())
+        .unique()
+        .sorted()
+        .collect::<HashSet<_>>();
+    let files_touched_by_diffs = base_diffs
+        .keys()
+        .map(|p| p.as_path())
+        .collect::<HashSet<_>>();
+
+    let touched_by_both = files_touched_by_commits
+        .intersection(&files_touched_by_diffs)
+        .map(|item| item.to_owned())
+        .map(|item| item.to_owned())
```

Comment:
It's unclear to me how I need to change `files_touched_by_commits` and `files_touched_by_diffs` to avoid this double `.to_owned()`.. 🤔

Reply:
What about *no* `to_owned()` ;)?

```patch
diff --git a/crates/gitbutler-branch-actions/src/status.rs b/crates/gitbutler-branch-actions/src/status.rs
index 18371cd24..605582862 100644
--- a/crates/gitbutler-branch-actions/src/status.rs
+++ b/crates/gitbutler-branch-actions/src/status.rs
@@ -289,8 +289,7 @@ fn compute_locks(
 
     let touched_by_both = files_touched_by_commits
         .intersection(&files_touched_by_diffs)
-        .map(|item| item.to_owned())
-        .map(|item| item.to_owned())
+        .copied()
         .collect::<Vec<_>>();
 
     let mut stacks_input: Vec<InputStack> = vec![];
@@ -302,7 +301,7 @@ fn compute_locks(
             let commit = repo.find_commit(commit_id)?;
             let files = list_virtual_commit_files(ctx, &commit, false)?;
             for file in files {
-                if touched_by_both.contains(&file.path) {
+                if touched_by_both.contains(&file.path.as_path()) {
                     let value = InputFile {
                         path: file.path,
                         diffs: file
```

Generally, `to_owned()` shouldn't be needed unless one wants to obtain ownership to return from the function, or similar.

Reply:
another thing i was gonna suggest - since what you use later in the function is PathBuf - you can skip the .as_path() and git it also a .cloned()